testsupport/wait/awaitility.go

```diff
@@ -732,6 +721,22 @@ func (w *Waiter[T]) FirstThat(predicates ...assertions.Predicate[client.Object])
 	return returnedObject, err
 }
 
+// WithNameMatching waits for a single object with the provided name in the namespace of the awaitality that additionally
+// matches the provided predicate function.
+func (w *Waiter[T]) WithNameMatching(name string, predicate func(T) bool) (T, error) {
+	return w.WithNameThat(name, &customPredicate[T]{
+		predicate: predicate,
+	})
+}
+
+type customPredicate[T client.Object] struct {
+	predicate func(T) bool
+}
+
+func (p *customPredicate[T]) Matches(object client.Object) bool {
```

Comment:
took me some time to figure this out, looked into the function WithNameThat to see why this was needed. maybe adding a comment on the function can help?